mos_tests/object_storage/test_crud_operations.py

```diff
@@ -0,0 +1,304 @@
+#    Copyright 2016 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import logging
+import math
+import re
+
+import pytest
+
+from mos_tests.neutron.python_tests.base import TestBase
+
+logger = logging.getLogger(__name__)
+
+
+class Helpers(object):
+
+    @staticmethod
+    def wa_for_1543135(env):
+        """WA for: Wrong OS_AUTH_URL makes keystone operations fail
+        https://bugs.launchpad.net/mos/+bug/1543135
+        Changes OS_AUTH_URL from ":5000/" --> ":5000/v2.0" in /root/openrc
+        """
+        logger.warning('Applying WA for bug #1543135')
+        cmd = r"sed -i 's#:5000/\x27#:5000/v2.0\x27#g' /root/openrc"
+        for controller in env.get_nodes_by_role('controller'):
+            with controller.ssh() as remote:
+                remote.check_call(cmd)
+
+
+@pytest.mark.undestructive
+@pytest.mark.check_env_('is_ceph_enabled')
+@pytest.mark.check_env_('is_radosgw_enabled')
+class TestObjectStorageS3CMD(TestBase, Helpers):
+    """Object Storage - CRUD Operations Tests
+    """
+    f_size = 100       # MB
+    f_size_b = 5432    # MB
+    chunk_size = 1024  # MB
+
+    @pytest.fixture(autouse=True)
+    def setUp_install_configure_s3cmd(
```

Comment:
Let's use one controller (without random) and make install action as class scoped fixture